File: server/src/main/java/com/logiflow/server/services/dispatch/OrderService.java
package com.logiflow.server.services.dispatch;

import com.logiflow.server.dtos.dispatch.OrderCreateRequest;
import com.logiflow.server.dtos.dispatch.OrderDto;
import com.logiflow.server.dtos.dispatch.OrderImportResponse;
import com.logiflow.server.dtos.dispatch.OrderListResponse;
import com.logiflow.server.dtos.dispatch.OrderUpdateRequest;
import org.springframework.web.multipart.MultipartFile;

public interface OrderService {
    OrderListResponse getOrders(String status, String date, int page, int size);
    OrderDto createOrder(OrderCreateRequest request, String username);
    OrderImportResponse importOrders(MultipartFile file, String username);
    byte[] downloadTemplate(String format);
    OrderDto getOrderById(Integer orderId);
    OrderDto updateOrder(Integer orderId, OrderUpdateRequest request);
}

File: client/src/services/dispatch/orderService.js
import api from '../api';

const basePath = '/dispatch';

const getOrders = async (params = {}) => {
  const response = await api.get(`${basePath}/orders`, { params });
  return response.data;
};

const createOrder = async (payload) => {
  const response = await api.post(`${basePath}/orders`, payload);
  return response.data;
};

const importOrders = async (file) => {
  const formData = new FormData();
  formData.append('file', file);
  const response = await api.post(`${basePath}/orders/import`, formData, {
    headers: { 'Content-Type': 'multipart/form-data' }
  });
  return response.data;
};

const downloadTemplate = async (format = 'csv') => {
  const response = await api.get(`${basePath}/orders/import/template`, {
    params: { format },
    responseType: 'blob'
  });
  return response.data; // blob
};

const getOrderById = async (orderId) => {
  const response = await api.get(`${basePath}/orders/${orderId}`);
  return response.data;
};

const updateOrder = async (orderId, payload) => {
  const response = await api.put(`${basePath}/orders/${orderId}`, payload);
  return response.data;
};

const getOrderTracking = async (orderId) => {
  const response = await api.get(`${basePath}/orders/${orderId}/tracking`);
  return response.data;
};

const updateOrderStatus = async (orderId, payload) => {
  const response = await api.put(`${basePath}/orders/${orderId}/status`, payload);
  return response.data;
};

const orderService = {
  getOrders,
  createOrder,
  importOrders,
  downloadTemplate,
  getOrderById,
  updateOrder,
  getOrderTracking,
  updateOrderStatus,
};

export default orderService;
